lib/job.js

```diff
@@ -217,17 +217,19 @@ class Job extends BaseModel {
         const pipelineFactory = PipelineFactory.getInstance();
         const jobFactory = JobFactory.getInstance();
 
-        const oldPeriodic = await jobFactory
-            .get(this.id)
-            .then(job => getAnnotations(job.permutations[0], 'screwdriver.cd/buildPeriodically'));
+        const oldJob = await jobFactory.get(this.id);
+        const oldPeriodic = getAnnotations(oldJob.permutations[0], 'screwdriver.cd/buildPeriodically');
 
         const newJob = await super.update();
         const pipeline = await pipelineFactory.get(newJob.pipelineId);
 
         try {
-            const newPeriodic = getAnnotations(this.permutations[0], 'screwdriver.cd/buildPeriodically');
+            const newPeriodic = getAnnotations(newJob.permutations[0], 'screwdriver.cd/buildPeriodically');
+            const isJobEnabled = newJob.state === 'ENABLED';
+            const isSettingUpdated = newPeriodic && newPeriodic !== oldPeriodic;
+            const wasJobDisabled = newPeriodic === oldPeriodic && oldJob.state === 'DISABLED';
 
-            if (newPeriodic && newPeriodic !== oldPeriodic) {
+            if ((isSettingUpdated || wasJobDisabled) && isJobEnabled) {
```

Comment:
shouldn't this be
```
const wasJobDisabled = oldJob.state === 'DISABLED';

 if ((isSettingUpdated || (wasJobDisabled && isJobEnabled)) { 


```

Reply:
But if, job is not enabled, then shouldn't it remove the periodic settings?

Reply:
won't then `isJobEnabled` will be false, and and then the following `if` block catches it ?